internal/namer/namer.go

```diff
@@ -229,9 +233,16 @@ func withScope(fqn, scope string) string {
 	return fqn + "/" + scope
 }
 
-// Sanitize replaces special characters in the string with underscores.
-func Sanitize(v string) string {
-	return invalidIdentifierChars.ReplaceAllLiteralString(v, "_")
+// sanitize replaces special characters in the string with underscores.
+// Before Cerbos 0.30 the names of resources or principals had to follow a certain pattern. We then replaced some of
+// the non-word characters with underscores because earlier versions of Cerbos used to generate Rego code for policies.
+// Because we used the sanitized name for computing the module ID of the policy, in order to maintain backward compatibility
+// and not break database stores we still have to do the same if the name matches the pattern.
+func sanitize(v string) string {
+	if oldNamePattern.MatchString(v) {
+		return invalidIdentifierChars.ReplaceAllLiteralString(v, "_")
+	}
+	return v
```

Comment:
Yes, unfortunately there will be some false positives. For disk-based stores, the `compile` stage would complain about finding duplicate policies. For DB stores, the insert would fail because of the upsert logic added in this PR.